pkg/utils/handle_errors.go

```diff
@@ -51,16 +53,16 @@ func HandleError(logger framework.Logger, c *gin.Context, err error) {
 		return
 	}
 
-	// will not captured by sentry if its an record not found error from gorm
+	// will not be captured by sentry if its a gorm.ErrRecordNotFound
 	if errors.Is(err, gorm.ErrRecordNotFound) {
 		c.JSON(http.StatusNotFound, gin.H{
-			"error": err.Error(),
+			"error": msgForUnhandledError,
 		})
 		return
 	}
```

Comment:
I think returning `api_errors.ErrRecordNotFound` is fine for this case instead of generic unhandled error. :)